src/mscorlib/src/System/StringComparer.cs

```diff
@@ -213,83 +213,77 @@ internal sealed class CultureAwareComparer : StringComparer
 
     }
 
-    // Provide x more optimal implementation of ordinal comparison.
+    // Provide a more optimal implementation of ordinal comparison.
     [Serializable]
     internal sealed class OrdinalComparer : StringComparer 
 #if FEATURE_RANDOMIZED_STRING_HASHING           
         , IWellKnownStringEqualityComparer
 #endif
     {
-        private bool            _ignoreCase;
-       
-        internal OrdinalComparer(bool ignoreCase) {
-               _ignoreCase = ignoreCase;
-        }
-
-        public override int Compare(string x, string y) {
-            if (Object.ReferenceEquals(x, y)) return 0;
-            if (x == null) return -1;
-            if (y == null) return 1;
-            
-            if( _ignoreCase) {
-                return String.Compare(x, y, StringComparison.OrdinalIgnoreCase);
-            }
-                    
-            return String.CompareOrdinal(x, y);                                
-        }
-                
-        public override bool Equals(string x, string y) {
-            if (Object.ReferenceEquals(x ,y)) return true;
-            if (x == null || y == null) return false;
+        public override int Compare(string x, string y) =>
+            object.ReferenceEquals(x, y) ? 0 :
+            x == null ? -1 :
+            y == null ? 1 :
+            string.CompareOrdinal(x, y);
 
-            if( _ignoreCase) {
-                if( x.Length != y.Length) {
-                    return false;
-                }
-                return (String.Compare(x, y, StringComparison.OrdinalIgnoreCase) == 0);                                            
-            }
-            return x.Equals(y);
-        }               
-                
-        public override int GetHashCode(string obj) {
-            if( obj == null) {
-                throw new ArgumentNullException(nameof(obj));
-            }
-            Contract.EndContractBlock();
+        public override bool Equals(string x, string y) =>
+            object.ReferenceEquals(x, y) ||
+            (x != null && x.Equals(y));
```

Comment:
Similarly, could this be `public override bool Equals(string x, string y) => string.Equals(x, y);`?